src/node_shapes.jl

```diff
@@ -0,0 +1,227 @@
+        sethue(text_color)
+        svg = get_latex_svg(text)
+        w, h = svgwh(svg)
+        text_dims = last(bounding_box)-first(bounding_box)
+        if text_align == :inside
+            scl = text_dims/(w*1.8, h*1.8)
+            scale(scl...)
+        else
+            scl = Point(1, 1)
+            clip ? clipreset() : nothing
+        end
+        # TODO: Work around for the frame gap between values bounding_box and text_box become available
+        if frame > first(get_frames(object))+1
+            latex(text, (text_box[1]+text_box[2])/(Tuple(scl).*2), :middle, :center)
+        end
+        scale(1/scl[1], 1/scl[2])
+        # Update for next frame
+        shift = (O, O)
+        if text_align == :top
+            shift = ((0, -text_dims[2]-offset), (0, -text_dims[2]-offset))
+        elseif text_align == :bottom
+            shift = ((0, text_dims[2]+offset), (0, text_dims[2]+offset))
+        elseif text_align == :left
+            shift = ((-text_dims[1]-offset, 0), (-text_dims[1]-offset, 0))
+        elseif text_align == :right
+            shift = ((text_dims[1]+offset, 0), (text_dims[1]+offset, 0))
+        end
+        object.meta.opts[:text_box] = bounding_box .+ shift
+    end
+    return opts, draw
+end
+
+"""
+    node_text(text::AbstractString, offset::Point, color::String)
+    node_text(text::LaTeXString, offset::Point, color::String)
+
+Label a node with text, which is also rotated to match the direction specified by `offset`.
+
+# Arguments
+- `offset::Point`: Specify the offset of the text label with respect to the node center.
+    - The text is translated and rotated in the direction of the offset.
+- `color`: Used only when `align` is not `:inside`. Specifies an offset in the direction of `align`.
+"""
+function node_text(text::AbstractString, offset::Point, color::String="black")
+    return (args... ; text=text, position=O, text_color=color, clip=false, kwargs...) -> begin
+        clip ? clipreset() : nothing
+        sethue(text_color)
+        translate(position+offset)
+        rotate(slope(O, offset)+pi/2)
+        Luxor.text(text, valign=:middle, halign=:center)
+        rotate(-slope(O, offset)-pi/2)
+        translate(-position-offset)
+    end
+end
+
+function node_text(text::LaTeXString, offset::Point, color::String="black")
+    return (args...; text=text, position=O, text_color=color, clip=false, kwargs...) -> begin
+        clip ? clipreset() : nothing
+        sethue(text_color)
+        translate(position+offset)
+        rotate(slope(O, offset)+pi/2)
+        latex(text, O, :middle, :center)
+        rotate(-slope(O, offset)-pi/2)
+        translate(-position-offset)
+    end
+end
+
+"""
+    node_fill(fill::Symbol, arg::String)
+
+Fill in the interior of the node with an image or a color.
+
+# Arguments
+- `fill::Symbol`: Can be `image` or `color`.
+- `arg::String`: Can be a color like "red" or a image path in the same directory like "image.png".
```

Comment:
In Luxor one normally sets colors with symbols so `:red` instead of "red"

Reply:
Hey, I was using `sethue` to switch between colors. I checked the methods available for `sethue` and could not find one using `Symbol`s. Is there something else you are referring to here?

Reply:
Oh sorry I think I got confused 😁 but I think other color packages are also supported. Let me check though

Reply:
See: https://juliagraphics.github.io/Luxor.jl/stable/colors-styles/